lib/web_ui/lib/src/engine/html_image_codec.dart

```diff
@@ -129,13 +129,13 @@ class HtmlImage implements ui.Image {
   final int height;
 
   @override
-  Future<ByteData?> toByteData(
-      {ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) {
-    return futurize((Callback<ByteData?> callback) {
-      return _toByteData(format.index, (Uint8List? encoded) {
-        callback(encoded?.buffer.asByteData());
-      });
-    });
+  Future<ByteData?> toByteData({ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) {
+    if (imgElement.src?.startsWith('data:') == true) {
+      final data = UriData.fromUri(Uri.parse(imgElement.src!));
+      return Future.value(data.contentAsBytes().buffer.asByteData());
+    } else {
+      return Future.value(null);
```

Comment:
I think this should probably throw instead of returning null. This indicates we don't support the format. WDYT?

Reply:
I really don't know what you normally do in a situation like that. If throw, than throw it is.